File: php/src/SimpleArmor.php
<?php

declare(strict_types=1);

namespace App;

class SimpleArmor implements Armor
{
    private int $soak;

    public function __construct(int $soak)
    {
        $this->soak = $soak;
    }

    public function getDamageSoak(): int
    {
        return $this->soak;
    }
}

File: go/simple_armor.go
package codingdojo

type SimpleArmor struct {
	soak int32
}

func MakeSimpleArmor(soak int32) SimpleArmor {
	return SimpleArmor{soak}
}

func (a SimpleArmor) GetDamageSoak() int32 {
	return a.soak
}
